configgtm-v1/asmap.go

```diff
@@ -0,0 +1,202 @@
+package configgtm
+
+import (
+	"github.com/akamai/AkamaiOPEN-edgegrid-golang/client-v1"
+)
+
+//
+// Handle Operations on gtm asmaps
+// Based on 1.3 schema
+//
+
+type AsAssignment struct {
+        DatacenterBase                                                    
+        AsNumbers                []int64           `json:"asNumbers"`
+}
+
+// AsMap  represents a GTM AsMap   
+type AsMap struct {
+        DefaultDatacenter        *DatacenterBase   `json:"defaultDatacenter"`
+        Assignments              []*AsAssignment   `json:"assignments,omitempty"`
+        Name                     string            `json:"name,omitempty"`
+        Links                    []*Link           `json:"links,omitempty"`
+}
+
+// NewAsMap creates a new asMap
+func NewAsMap(asmapname string) *AsMap {
+	asmap := &AsMap{Name: asmapname}
+        return asmap
+}
+
+// GetAsMap retrieves a asMap with the given name.                     
+func GetAsMap(asmapname, domainname string) (*AsMap, error) {
+	as := NewAsMap(asmapname)
+	req, err := client.NewRequest(
+		Config,
+		"GET",
+		"/config-gtm/v1/domains/"+domainname+"/as-maps/"+asmapname,
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+        SetHeader(req)
+
+        printHttpRequest(req, true)
+
+	res, err := client.Do(Config, req)
+	if err != nil {
+		return nil, err
+	}
+
+        printHttpResponse(res, true)
+
+	if client.IsError(res) && res.StatusCode != 404 {
+		return nil, client.NewAPIError(res)
+	} else if res.StatusCode == 404 {
+		return nil, &CommonError{entityName: "asMap", name: asmapname}
+	} else {
+		err = client.BodyJSON(res, as)
+		if err != nil {
+			return nil, err
+		}
+
+		return as, nil
+	}
+}
+
+// Instantiate new Assignment struct
+func (as *AsMap) NewAssignment(dcid int, nickname string) *AsAssignment {
+        asAssign := &AsAssignment{}
+        asAssign.DatacenterId = dcid 
+        asAssign.Nickname =nickname
+
+        return asAssign
+
+}
+
+// Instantiate new Default Datacenter Struct
+func (as *AsMap) NewDefaultDatacenter(dcid int) *DatacenterBase {
+        return &DatacenterBase{DatacenterId: dcid}
+}
+
+// Create asMap in provided domain                        
+func (as *AsMap) Create(domainname string) (*AsMapResponse, error) {
+
+        // Use common code. Any specific validation needed?
+
+        return as.save(domainname)
+
+}
+
+// Update AsMap in given domain
+func (as *AsMap) Update(domainname string) (*ResponseStatus, error) {
+
+        // common code
+   
+        stat, err := as.save(domainname)
+        if err != nil {
+                return nil, err
+        }
+        return stat.Status, err
+
+}
+
+// Save AsMap in given domain. Common path for Create and Update.
+func (as *AsMap) save(domainname string) (*AsMapResponse, error) {
```

Comment:
`Save`

Reply:
save is intended to be internal and not exposed. Used by Create and Update methods.